Adds menu visibility logic

joaofreitas-qubit opened the pull request:
Relates with: #qubIT-Omnis-6533

joaofreitas-qubit committed:
Adds menu visibility logic

Relates with: #qubIT-Omnis-6533

joaofreitas-qubit commented on bennu-portal/src/main/java/org/fenixedu/bennu/portal/servlet/PortalInitializer.java:
```diff
@@ -55,13 +54,40 @@ public void contextInitialized(ServletContextEvent sce) {
+         * #qubIT-Omnis-6137
+         *
+         */
+        migrateMenuItemVisibility(PortalConfiguration.getInstance().getMenu());
```
@pabrantes although i performed this migration locally and I still had 13 results when i ran the query

select * from MENU_ITEM WHERE VISIBILITY IS NULL AND VISIBLE IS NOT NULL\G

Some of them didn't have parents so probably they are roots for the reamining ones and the dfs couldn't visit them because of that. Will this create a problem?

pabrantes replied:
Well we need to understand if they are just another unreachable tree. If so we're safe, otherwise if they are reachable somehow the transactions will fail because the required consistency predicate will throw an exception. Ping me on monday so we can look into the data and see what are those items.

joaofreitas-qubit committed:
Fixes menu visibility migration

Relates with: #qubIT-Omnis-6531
The problem was that I was using orderedchilds instead of childset.
The comparator in the treeset identified the entries as equal if the
result of the compareTo was the same so some entries were not being migrated
